templates/commands/render/manifest.go

```diff
@@ -0,0 +1,214 @@
+	src string
+
+	// The temp directory where the template was downloaded.
+	templateDir string
+}
+
+// canonicalSourcer is the subset of the Downloader interface that is needed by
+// the manifest code.
+type canonicalSourcer interface {
+	CanonicalSource(_ context.Context, cwd, destDir string) (string, bool, error)
+}
+
+// writeManifest creates a manifest struct, marshals it as YAML, and writes it
+// to destDir/.abc/ .
+func writeManifest(ctx context.Context, p *writeManifestParams) (rErr error) {
+	canonicalSource, _, err := p.cs.CanonicalSource(ctx, p.cwd, p.destDir)
+	if err != nil {
+		return err //nolint:wrapcheck
+	}
+
+	m, err := buildManifest(ctx, p, canonicalSource)
+	if err != nil {
+		return err
+	}
+
+	buf, err := yaml.Marshal(m)
+	if err != nil {
+		return fmt.Errorf("failed marshaling Manifest when writing: %w", err)
+	}
+
+	filename, err := newManifestFilename(p, canonicalSource)
+	if err != nil {
+		return err
+	}
+
+	if p.dryRun {
+		if _, err := p.fs.Stat(filename); err != nil {
+			if common.IsStatNotExistErr(err) {
+				// This is good. We don't want to overwrite an existing manifest file,
+				// so that fact that it doesn't already exist is good news.
+				return nil
+			}
+			return fmt.Errorf("Stat(): %w", err)
+		}
+		return fmt.Errorf("dry run failed, the output manifest file %q already exists", filename)
+	}
+
+	// Why O_EXCL? Because we don't want to overwrite an existing file. TODO test
+	fh, err := p.fs.OpenFile(filename, os.O_CREATE|os.O_EXCL|os.O_WRONLY, common.OwnerRWPerms)
+	if err != nil {
+		return fmt.Errorf("OpenFile(%q): %w", filename, err)
+	}
+	defer func() {
+		rErr = errors.Join(rErr, fh.Close())
+	}()
+	buf = append([]byte("# Generated by the \"abc templates\" command. Do not modify.\n"), buf...)
+	if _, err := fh.Write(buf); err != nil {
+		return fmt.Errorf("Write(%q): %w", filename, err)
+	}
+
+	return nil
+}
+
+// newManifestFilename outputs the filename that will be used for a newly rendered
+// template (not an upgrade to an already-installed manifest). This includes the
+// ".abc/" prefix.
+func newManifestFilename(p *writeManifestParams, canonicalSource string) (string, error) {
+	manifestDir := filepath.Join(p.destDir, manifestDirName)
+	if err := p.fs.MkdirAll(manifestDir, common.OwnerRWXPerms); err != nil {
+		return "", fmt.Errorf("failed creating %s directory to contain manifest: %w", manifestDir, err)
+	}
+
+	namePart := "nolocation"
+	if canonicalSource != "" {
+		namePart = url.PathEscape(canonicalSource)
+	}
+
+	// We include the creation time in the filename to disambiguate between
+	// multiple installations of the same template that target the same
+	// destination directory.
+	timeStr := p.clock.Now().UTC().Format(time.RFC3339Nano)
+
+	// Windows doesn't allow colons in filenames 🙄
+	timeStr = strings.ReplaceAll(timeStr, ":", "_")
+
+	baseName := strings.Join(
+		[]string{"manifest", namePart, timeStr},
+		"_")
+	baseName += ".lock.yaml"
```

Comment:
I'm not so sure, it would look like
```
baseName := strings.Join(
		[]string{"manifest", namePart, timeStr},
		"_") + ".lock.yaml"
```

which seems a bit complex for a one-liner.

Reply:
Actually, that would be fine. Updated.